lib/matplotlib/rcsetup.py

```diff
@@ -1012,9 +1006,9 @@ def _convert_validator_spec(key, conv):
     # If "data", axes limits are set close to the data.
     # If "round_numbers" axes limits are set to the nearest round numbers.
     "axes.autolimit_mode": ["data", "round_numbers"],
-    "axes.xmargin": _range_validators["0 <= x <= 1"],  # margin added to xaxis
-    "axes.ymargin": _range_validators["0 <= x <= 1"],  # margin added to yaxis
-    'axes.zmargin': _range_validators["0 <= x <= 1"],  # margin added to zaxis
+    "axes.xmargin": _validate_greaterthan_minushalf,  # margin added to xaxis
+    "axes.ymargin": _validate_greaterthan_minushalf,  # margin added to yaxis
+    'axes.zmargin': _validate_greaterthan_minushalf,  # margin added to zaxis
```

Comment:
For consistency with the other lines
```suggestion
    "axes.zmargin": _validate_greaterthan_minushalf,  # margin added to zaxis
```

Reply:
yup, fixed.